openedx/features/smart_referral/helpers.py

```diff
@@ -57,3 +59,23 @@ def get_email_domain(email):
     Return email domain name e.g "google.com", "yahoo.com" and "outlook.com" etc by splitting email after "@".
     """
     return email.split('@')[1] if email else None
+
+
+def filter_referred_contacts(contacts, user):
+    """
+    Filter out contacts which are already been referred twice or which are referred by current user.
+    """
+    if not contacts:
+        return None
+
+    filtered_contacts = []
+    for contact in contacts:
+        qs_referral = SmartReferral.objects.filter(contact_email=contact.get('contact_email'))
+        qs_count = qs_referral.count()
+
+        # add contact to list if it has never been referred before or if it was referred
+        # only once by any other user (not by current user)
+        is_referred_before = qs_count > 1 or (qs_count == 1 and qs_referral.first().user == user)
+        filtered_contacts.append(contact) if not is_referred_before else None
```

Comment:
how about?
```
if qs_count == 0 or qs_reffered.first().user != user:
    filtered_contacts.append(contact)
```

Now if we give qs_count and qa_reffered some more meaningful names, we don't even need to add the comment to explain the logic.

Reply:
This condition will not work, we need to cater following scenarios:

1. If `qs_count` >=2, `is_referred_before` must return True
2. If `qs_count` is exactly 1 and `first().user` is same as current user then `is_referred_before ` must return True
3. If `qs_count` == 0, `is_referred_before` must return Flase

Reply:
Just forget about is_reffered_before, consider actual criteria you have mentioned in the code and lemme know if any case is not tackled by this condition.